### DISCUSSION TITLE
About BITMAPFILEHEADER class

### DISCUSSION THREAD

**Discussion by H1manshu21** [Blaze-Stars/Image-Filter-Application#37] [Q&A] [answered]
About to discuss in future.

**Comment by Ank1taS** [comment, marked as answer]
The BITMAPFILEHEADER structure contains information about **BMP file** i.e the type, size, and layout of a file that includes a DIB. You can refer to the [Microsoft documentation](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-bitmapfileheader). and about [file format](https://en.wikipedia.org/wiki/BMP_file_format)

### Syntax

```c++

struct tagBITMAPFILEHEADER {
  WORD  bfType;
  DWORD bfSize;
  WORD  bfReserved1;
  WORD  bfReserved2;
  DWORD bfOffBits;
} ;
```

BMP Header ```BITMAPFILEHEADER```  contains a total of **5 variables** 
| Variable | Field Name | Size in bytes | Description | Reference |
| -------- | ---------- | ------------- | ----------- | --------- |
|  bfType | FileType  | 2 bytes  |  A 2 character string value in ASCII to specify a DIB file type. It must be **'BM'** or **'0x42 0x4D'** in hexadecimal for modern compatibility reasons. | [This for details](https://stackoverflow.com/questions/601430/multibyte-character-constants-and-bitmap-file-header-type-constants)|   
|bfSize|  FileSize | 4 bytes  |  An integer (unsigned) representing entire **file size in bytes**. This value is basically the **number of bytes in a BMP image file**. |  | 
|bfReserved1| Reserved  |  2 bytes |  These 2 bytes are reserved to be utilized by an image processing application to add additional meaningful information. It should be **initialized to a '0' integer (unsigned) value**. | [This for details](https://stackoverflow.com/questions/4294304/what-is-rgbreserved)|  |
|bfReserved2| Reserved  |  2 bytes |  These 2 bytes are reserved to be utilized by an image processing application to add additional meaningful information. It should be **initialized to a '0' integer (unsigned) value**. | [This for details](https://stackoverflow.com/questions/4294304/what-is-rgbreserved). | 
| bfOffBits | PixelDataOffset  |  4 bytes | An integer (unsigned) representing the offset of actual pixel data in bytes. i.e **the number of bytes between the start of the file (0) and the first byte of the pixel data.**  |  | 


In this way **size of BITMAPFILEHEADER is _14 bytes_**
